test/integration/server-selection/server_selection.prose.operation_count.test.ts

```diff
@@ -27,19 +26,12 @@ async function runTaskGroup(collection: Collection, count: 10 | 100 | 1000) {
 
 async function ensurePoolIsFull(client: MongoClient) {
   let connectionCount = 0;
-  const onConnectionCreated = () => connectionCount++;
-  client.on('connectionCreated', onConnectionCreated);
-
-  // 250ms should be plenty of time to fill the connection pool,
-  // but just in case we'll loop a couple of times.
-  for (let i = 0; connectionCount < POOL_SIZE * 2 && i < 10; ++i) {
-    await promisify(setTimeout)(250);
-  }
-
-  client.removeListener('connectionCreated', onConnectionCreated);
-
-  if (connectionCount !== POOL_SIZE * 2) {
-    throw new Error('Connection pool did not fill up');
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for await (const _event of on(client, 'connectionCreated')) {
```

Comment:
I like the refactor but I originally added the error to make it more clear what happened if the pool ever failed to fill up and the test timed out.

Maybe we could race this function against a timer if a better error message is something we care about?

Reply:
I can update the naming according to what we discussed, but I don't think the error here is important. If it times out before the beforeEach is satisfied, it times out; a beforeEach isn't the right place to assert on how long something should take, and, imho, this isn't really different from awaiting any other precondition in a beforeEach (e.g., connecting to a server)